### PR TITLE
Update/refactor integrations/user-management-api/buyers

### PR DESCRIPTION
**Pull request by josemigallas**
♻️ refactor a lot
✅ fixes deprecation params as positional arg
🚨 fixes lint (rubocop)
🎨 improves code format

### CODE REVIEW

File: test/integration/user-management-api/buyers_account_plan_change_test.rb [hunk 1]
```diff
@@ -1,63 +1,84 @@
...
+    test 'change inexistent account plan' do
+      put change_plan_admin_api_account_path(@buyer, provider_key: @provider.api_key,
+                                                     plan_id: 'wololo',
+                                                     format: :xml)
+      assert_xml_404
+    end
 
-    host! @provider.admin_domain
-  end
+    context 'published plan' do
```
Comment (hallelujah) [inline test/integration/user-management-api/buyers_account_plan_change_test.rb:26]: please do not use context anymore, they are the root cause of random failures

File: test/integration/user-management-api/buyers_account_plan_change_test.rb [hunk 2]
```diff
@@ -1,63 +1,84 @@
...
 
-  test 'successful change account plan' do
-    put change_plan_admin_api_account_path(@buyer,
-                                                :provider_key => @provider.api_key,
-                                                "plan_id" => @published_account_plan.id,
-                                                :format => :xml)
+      attr_reader :plan
 
-    assert_response :success
+      should 'change account plan to a non-published one is permitted' do
```
Comment (hallelujah) [inline test/integration/user-management-api/buyers_account_plan_change_test.rb:33]: Please use `test` do we are going to get rid of shoulda-context

**Review by hallelujah — CHANGES_REQUESTED** [review]

File: test/integration/user-management-api/buyers_application_referrer_filters_test.rb [hunk 3]
```diff
@@ -27,31 +29,31 @@ def setup
...
+    get(admin_api_account_application_referrer_filters_path(@buyer, app), params: { access_token: token.value })
     assert_response :success
   end
 
   #TODO: clean the tests
   test 'create referrer filters' do
     application = @buyer.bought_cinstances.last
-    application_id = application.application_id
-    application.service.update_attribute :referrer_filters_required, true
+    application.application_id
```
Comment (akostadinov) [inline test/integration/user-management-api/buyers_application_referrer_filters_test.rb:42]: What is the point of this repeated statement?

Comment (akostadinov) [inline test/integration/user-management-api/buyers_application_referrer_filters_test.rb:42]: What is the point of this repeated statement? Not that I make any sense of the original.

**Review by akostadinov — COMMENTED** [review]
Looks nice. Just one question.

File: test/integration/user-management-api/buyers_application_referrer_filters_test.rb (hunk 3 above)
Comment (josemigallas) [reply test/integration/user-management-api/buyers_application_referrer_filters_test.rb:42]: It makes no sense at all

**Review by jsmadis — DISMISSED** [review]

File: test/integration/user-management-api/buyers_account_plan_change_test.rb [hunk 4]
```diff
@@ -1,63 +1,62 @@
...
-                                                "plan_id" => @published_account_plan.id,
-                                                :format => :xml)
-
+                                           provider_key: @provider.api_key,
+                                           plan_id: @published_account_plan.id,
+                                           format: :xml)
     assert_response :success
 
-    assert @buyer.bought_account_plan == @published_account_plan
+    assert_equal @buyer.bought_account_plan, @published_account_plan
```
Comment (hallelujah) [inline test/integration/user-management-api/buyers_account_plan_change_test.rb:27]: See https://github.com/3scale/porta/pull/2646/files#r752312663

Comment (josemigallas) [reply test/integration/user-management-api/buyers_account_plan_change_test.rb:27]: Right, can we let this one pass though?

**Review by akostadinov — APPROVED** [review]
I think this covers all concerns raised.

**Review by jsmadis — APPROVED** [review]

**Pull request merged**